1_data_prep/13_tree_crops.py

```diff
@@ -0,0 +1,353 @@
+        unique_mask_values = np.unique(mask_ids) # get unique mask ids
+
+        # load the json file that maps the mask IDs to their unique_ID values and create a mapping dict
+        IDs_to_labels_path = src_info["IDs_to_labels_path"]
+        with open(IDs_to_labels_path, "r") as f:
+            IDs_to_labels = json.load(f)
+        ID_mapping = {int(k): int(v) for k, v in IDs_to_labels.items()}
+
+        if LABELLED_ONLY:
+            # filter mask values to only those that map to labelled IDs
+            unique_mask_values = [uid for uid in unique_mask_values 
+                                  if ID_mapping.get(int(uid)) in labelled_tree_ids]
+
+        # iterate over ids
+        for tree_mask_value in unique_mask_values:
+            if np.isnan(tree_mask_value): # skip nan values
+                continue
+
+            # map mask value -> tree unique ID
+            tree_unique_id = ID_mapping.get(int(tree_mask_value))
+            if tree_unique_id is None:
+                raise ValueError(f"Tree unique_ID for mask value {tree_mask_value} is None. Check {IDs_to_labels_path}")
+
+            tree_unique_id = str(int(tree_unique_id)).zfill(5) # convert tree uid to 0 padded str to match plot_attributes format
+
+            # see if this tree has a species label
+            species_val = plot_attributes.loc[plot_attributes['unique_ID'] == tree_unique_id, 'species_code']
+            if len(species_val) == 0:
+                original_species = None
+            else:
+                original_species = species_val.iloc[0]
+                if pd.isna(original_species):
+                    original_species = None
+
+            mapping_stats['total_processed'] += 1
+
+            # Handle species mapping
+            if original_species is not None:
+                mapped_species, is_mapped = map_species(original_species, species_mapping, MAPPING_LEVEL)
+                
+                if is_mapped:
+                    mapping_stats['mapped_species'] += 1
+                    final_species = mapped_species
+                    should_save_crop = True
+                else:
+                    mapping_stats['unmapped_species'] += 1
+                    if INCLUDE_UNMAPPED_SPECIES:
+                        final_species = original_species  # Keep original species
+                        should_save_crop = True
+                    else:
+                        mapping_stats['skipped_unmapped'] += 1
+                        continue  # Skip this tree entirely
+            else:
+                # No species label
+                final_species = None
+                should_save_crop = True
+                is_mapped = False
+
+            # Determine file naming and directory structure
+            species_file_label = 'NONE' if final_species is None else str(final_species)
+            species_dir_label = 'unlabelled' if final_species is None else 'labelled'
+
+            if not should_save_crop:
```

Comment:
nit: seems like `should_save_crop` is true in all control flow branches where you haven't hit `continue`.